### PR TITLE
Add is_composite for factorial function fix 8724

### LINKED ISSUE #8724: If n is a nonnegative integer, factorial(n+3) should be known to be composite
Currently, if `n` is a nonnegative integer, `factorial(n+3).is_composite` returns `None`.

### PR DESCRIPTION
**Pull request by aktech**
Currently If n is a nonnegative integer, factorial(n+3).is_composite returns None.

This PR Fixes this issue.
Fix #8724

### CODE REVIEW

**Comment by aktech** [comment]
@skirpichev  @pelegm 
Please have a look.

File: sympy/functions/combinatorial/factorials.py [hunk 1]
```diff
@@ -172,6 +172,13 @@ def _eval_is_positive(self):
         if self.args[0].is_integer and self.args[0].is_nonnegative:
             return True
 
+    def _eval_is_composite(self):
+        x = self.args[0]
+        if x.is_integer:
+            return (x - 3).is_nonnegative
+        elif x.is_nonpositive or x.is_noninteger:
+            return False
```
Comment (skirpichev) [inline sympy/functions/combinatorial/factorials.py]: That's just wrong.

Comment (skirpichev) [inline sympy/functions/combinatorial/factorials.py]: On Mon, Jan 05, 2015 at 12:14:54PM -0800, AMiT Kumar wrote:
>    I didn't considered negatives.

Actually, I'm not just about negatives.

**Comment by aktech** [comment]
@skirpichev 
Fixed Logic : 
factorial(x) will return composite if `x >= 3` (x : integer)
Since `3! = 3x2x1` is composite, and so 4! , 5! , 6! .....and so on.

So (x - 3).is_nonnegative will return True when 
`x - 3 >= 0`  which means `x >= 3` (positive integer > 3).

It will return False is `x - 3 <= 0` which means `x <= -3` negative Integer (x : Integer)
and for every negative integer factorial (x) is zoo.

Otherwise it will return None. (which means it can't be assumed from given info)

**Comment by skirpichev** [comment]
+1

please rebase pr (squash second commit and fix commit msg)

**Comment by aktech** [comment]
Rebased.
Thanks!

**Comment by pelegm** [comment]
+1 (if passes tests)

**Pull request merged**